File: CodingQuestions/hw40/mongo-blog/src/main/java/com/chuwa/mongoblog/service/PostService.java
package com.chuwa.mongoblog.service;



import com.chuwa.mongoblog.payload.PostDto;

import java.util.List;

public interface PostService {
    PostDto createPost(PostDto postDto);
    List<PostDto> getAllPost();

     PostDto getPostById(long id);
     PostDto updatePostById(PostDto postDto, long id);
     void removePostById(long id);
}

File: MavenProject/KotlinRedbook/src/main/kotlin/com/chuwa/KotlinRedbook/Service/PostService.kt
package com.chuwa.KotlinRedbook.Service

import com.chuwa.KotlinRedbook.DAO.PostRepository
import com.chuwa.KotlinRedbook.Entity.Post
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class PostService {
    @Autowired
    lateinit var postRepository : PostRepository

    fun createPost(title : String, content : String?) : Post {
        var newPost : Post = Post(title, content)
        var saved : Post = postRepository.save(newPost)
        return saved
    }

    fun getPostById(id : Long) : Post? {
        var result = postRepository.findById(id).orElse(null)
        return result
    }

    fun getAllPost() : List<Post> {
        return postRepository.findAll()
    }
}